Parser/pegen/pegen.c

```diff
@@ -2160,3 +2160,94 @@ _PyPegen_nonparen_genexp_in_call(Parser *p, expr_ty args)
         "Generator expression must be parenthesized"
     );
 }
+
+OperatorTermPair *
+_PyPegen_operator_term_pair(Parser *p, Token* oper, expr_ty factor)
+{
+    OperatorTermPair *a = PyArena_Malloc(p->arena, sizeof(OperatorTermPair));
+    if (!a) {
+        return NULL;
+    }
+    a->oper = oper;
+    a->factor = factor;
+    return a;
+}
+
+static operator_ty
+_PyPegen_get_binary_operator(const Token *t)
+{
+    switch (t->type) {
+        case VBAR:
+            return BitOr;
+        case CIRCUMFLEX:
+            return BitXor;
+        case AMPER:
+            return BitAnd;
+        case LEFTSHIFT:
+            return LShift;
+        case RIGHTSHIFT:
+            return RShift;
+        case PLUS:
+            return Add;
+        case MINUS:
+            return Sub;
+        case STAR:
+            return Mult;
+        case AT:
+            return MatMult;
+        case SLASH:
+            return Div;
+        case DOUBLESLASH:
+            return FloorDiv;
+        case PERCENT:
+            return Mod;
+        default:
+            return (operator_ty)0;
+    }
+}
+
+expr_ty
+_PyPegen_operator_precedence_expr(Parser *p, expr_ty factor, asdl_seq *terms) {
+    expr_ty result = NULL;
+
+    expr_ty *output= PyMem_Calloc(asdl_seq_LEN(terms) + 1, sizeof(expr_ty));
+    Token **stack = PyMem_Calloc(asdl_seq_LEN(terms), sizeof(Token*));
```

Comment:
The names `output` and `stack` don't convey much meaning -- they both are part of the "stack". Maybe just `values` and `operators`? And a comment explaining how they relate to each other.